config/src/main/java/org/springframework/security/config/annotation/web/configuration/HttpSecurityConfiguration.java

```diff
@@ -226,23 +226,15 @@ private PasswordEncoder getPasswordEncoder() {
 			if (this.passwordEncoder != null) {
 				return this.passwordEncoder;
 			}
-			PasswordEncoder passwordEncoder = getBeanOrNull(PasswordEncoder.class);
+			PasswordEncoder passwordEncoder = this.applicationContext.getBeanProvider(PasswordEncoder.class)
+				.getIfUnique();
 			if (passwordEncoder == null) {
 				passwordEncoder = PasswordEncoderFactories.createDelegatingPasswordEncoder();
 			}
 			this.passwordEncoder = passwordEncoder;
 			return passwordEncoder;
 		}
 
-		private <T> T getBeanOrNull(Class<T> type) {
```

Comment:
I think this should keep `ObjectProvider#getIfUnique` beacause `ObjectProvider#getIfAvailable` will throw `NoUniqueBeanDefinitionException` if there is more than one bean, but `ObjectProvider#getIfUnique` will return `null` value instead

Reply:
That's correct, which is why we want to use `getIfAvailable`. Because the pre-existing approach would throw an exception if not unique (this is what `getBean` does), then we should preserve that behavior.

Reply:
I see `getBean` will return null if it throws `NoSuchBeanDefinitionException`
In case use `getIfAvailable` and there is more than one bean, it will throw `NoUniqueBeanDefinitionException`. But
`NoUniqueBeanDefinitionException extends NoSuchBeanDefinitionException`, so I think we should use `getIfUnique` instead.

Reply:
Ah, thank you for pointing that out, I appreciate that. I agree, let's use `getIfUnique` in these places since by catching `NoSuchBeanDefinitionException`, it will return `null` if no bean or if a duplicate bean.